airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2109,3 +2122,287 @@ def execute(self, context: Any):
+                }
+            },
+            config_variables_file_path=None,
+            commented_map=None,
+        )
+        data_context = BaseDataContext(project_config=project_config)
+
+        return data_context
+
+    def get_batch_kwargs(self):
+
+        # Tell GE where to fetch the batch of data to be validated.
+        batch_kwargs = {
+            "datasource": "bq_datasource",
+        }
+
+        if self.validation_type == GreatExpectationsValidations.SQL.value:
+            batch_kwargs["query"] = self.validation_type_input
+            batch_kwargs["data_asset_name"] = self.bq_dataset_name
+            batch_kwargs["bigquery_temp_table"] = self.get_temp_table_name()
+        elif self.validation_type == GreatExpectationsValidations.TABLE.value:
+            batch_kwargs["table"] = self.validation_type_input
+            batch_kwargs["data_asset_name"] = self.bq_dataset_name
+
+        log.info("batch_kwargs: " + str(batch_kwargs))
+
+        return batch_kwargs
+
+    def execute(self, context):
+
+        data_context = self.instantiate_data_context()
+
+        log.info("Loading expectations...")
+        suite = data_context.get_expectation_suite((self.expectations_file_name.rsplit(".", 1)[0]))
+
+        batch_kwargs = self.get_batch_kwargs()
+
+        log.info("Getting the batch of data to be validated...")
+        batch = data_context.get_batch(batch_kwargs, suite)
+
+        run_id = {
+            "run_name": 'bq',
+            "run_time": datetime.datetime.now(datetime.timezone.utc)
+        }
+
+        log.info("Validating batch against expectations...")
+        results = data_context.run_validation_operator(
+            "action_list_operator",
+            assets_to_validate=[batch],
+            run_id=run_id)
+
+        validation_result_identifier = list(results['run_results'].keys())[0]
+        # For the given validation_result_identifier, get a link to the data docs that were generated by Great
+        # Expectations as part of the validation.
+        data_docs_url = \
+            data_context.get_docs_sites_urls(resource_identifier=validation_result_identifier, site_name='GCS_site')[0][
+                'site_url']
+        log.info("Data docs url is: %s", data_docs_url)
+        if results["success"]:
+            self.log.info('All expectations met')
+        else:
+            self.log.info('One or more expectations were not met.')
+            if self.send_alert_email:
+                self.log.info('Sending alert email...')
+                self.send_alert(data_docs_url)
+            if self.fail_if_expectations_not_met:
+                raise AirflowException('One or more expectations were not met')
+
+    # Generate a unique name for a temporary BQ table
+    def get_temp_table_name(self):
+        now = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+        name_start = "temp_great_expectations_" + now + '_'
+        full_name = name_start + ''.join(str(uuid.uuid4().hex))
```

Comment:
@turbaszek , as suggested, changed the temp table name to be generated based on a uuid.